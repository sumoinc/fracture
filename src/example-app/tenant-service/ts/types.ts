export type Error = {
  code: number;
  source: string;
  message: string;
  detail: string;
}

export type Response<T> = {
  data?: T;
  errors: Error[];
  status: number;
}

export type ListResponse<T> = {
  data?: T[];
  errors: Error[];
  status: number;
}

/**
 * A gereric type
 */
export interface Tenant {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface TenantMessage {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface CreateTenantInput {
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface CreateTenantOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface GetTenantInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface GetTenantOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface UpdateTenantInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface UpdateTenantOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface DeleteTenantInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface DeleteTenantOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A name.
   */
  name: string;
  /**
   * A nickname.
   */
  nickname?: string;
}

/**
 * A gereric type
 */
export interface ListTenantInput {
}

/**
 * A gereric type
 */
export interface ListTenantOutput {
}

/**
 * A gereric type
 */
export interface Actor {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface ActorMessage {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface CreateActorInput {
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface CreateActorOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface GetActorInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface GetActorOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface UpdateActorInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface UpdateActorOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface DeleteActorInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface DeleteActorOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName: string;
}

/**
 * A gereric type
 */
export interface ListActorInput {
}

/**
 * A gereric type
 */
export interface ListActorOutput {
}

// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
