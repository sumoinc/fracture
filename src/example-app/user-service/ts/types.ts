export type Error = {
  code: number;
  source: string;
  message: string;
  detail: string;
}

export type Response<T> = {
  data?: T;
  errors: Error[];
  status: number;
}

export type ListResponse<T> = {
  data?: T[];
  errors: Error[];
  status: number;
}

/**
 * A gereric type
 */
export interface User {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface UserMessage {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface CreateUserInput {
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface CreateUserOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface GetUserInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface GetUserOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface UpdateUserInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface UpdateUserOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface DeleteUserInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface DeleteUserOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A first-name.
   */
  firstName: string;
  /**
   * A last-name.
   */
  lastName?: string;
}

/**
 * A gereric type
 */
export interface CreateUserVersionInput {
}

/**
 * A gereric type
 */
export interface CreateUserVersionOutput {
}

/**
 * A gereric type
 */
export interface GetUserVersionInput {
}

/**
 * A gereric type
 */
export interface GetUserVersionOutput {
}

/**
 * A gereric type
 */
export interface Group {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface GroupMessage {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface CreateGroupInput {
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface CreateGroupOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface GetGroupInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface GetGroupOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface UpdateGroupInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface UpdateGroupOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface DeleteGroupInput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
}

/**
 * A gereric type
 */
export interface DeleteGroupOutput {
  /**
   * The id for the record.
   * @readonly This attribute is managed automatically by the system.
   */
  id: string;
  /**
   * The type for this record.
   * @readonly This attribute is managed automatically by the system.
   */
  type: string;
  /**
   * The version of this record
   * @default "LATEST"
   * @readonly This attribute is managed automatically by the system.
   */
  version: string;
  /**
   * The date and time this record was created.
   * @readonly This attribute is managed automatically by the system.
   */
  createdAt: string;
  /**
   * The date and time this record was last updated.
   * @readonly This attribute is managed automatically by the system.
   */
  updatedAt: string;
  /**
   * The date and time this record was deleted.
   * @readonly This attribute is managed automatically by the system.
   */
  deletedAt: string;
  /**
   * A name.
   */
  name: string;
}

/**
 * A gereric type
 */
export interface CreateGroupVersionInput {
}

/**
 * A gereric type
 */
export interface CreateGroupVersionOutput {
}

/**
 * A gereric type
 */
export interface GetGroupVersionInput {
}

/**
 * A gereric type
 */
export interface GetGroupVersionOutput {
}

// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
