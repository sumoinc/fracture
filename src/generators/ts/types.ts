import { buildTypes } from "./factories/build-type";
import { printNodes } from "./factories/print-nodes";
import { FractureService } from "../../core/fracture-service";
import { GeneratedTypescriptFile } from "../generated-typescript-file";

export class GenerateTypes extends GeneratedTypescriptFile {
  constructor(service: FractureService) {
    super(service, "types.ts");
  }

  preSynthesize(): void {
    super.preSynthesize();

    /***************************************************************************
     * Service Level Types
     **************************************************************************/

    const service = this.project as FractureService;
    const serviceTypes = buildTypes({
      service,
      structures: service.structures,
    });

    // this loop seems useless but it creates empty lines between each type.
    serviceTypes.forEach((type) => {
      this.addLine(printNodes([type]));
    });

    service.resources.forEach((resource) => {
      // baseline data entities
      const resourceTypes = buildTypes({
        service,
        structures: resource.structures,
      });
      // this loop seems useless but it creates empty lines between each type.
      resourceTypes.forEach((type) => {
        this.addLine(printNodes([type]));
      });

      // inputs and outputs for each operation
      resource.operations.forEach((operation) => {
        const operationTypes = buildTypes({
          service,
          structures: operation.structures,
        });

        // this loop seems useless but it creates empty lines between each type.
        operationTypes.forEach((type) => {
          this.addLine(printNodes([type]));
        });
      });
    });
  }
}
