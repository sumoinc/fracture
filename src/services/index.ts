export * from "./data-service";
export * from "./operation";
export * from "./resource-attribute";
export * from "./resource";
export * from "./service";
export * from "./structure-attribute";
export * from "./structure";
