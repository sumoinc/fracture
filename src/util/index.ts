export * from "./determineBuildId";
export * from "./test-util";
