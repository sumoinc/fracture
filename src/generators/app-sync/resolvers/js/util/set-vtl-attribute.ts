import { ResourceAttributeGenerator } from "../../../../../services/resource-attribute";
import { StructureAttribute } from "../../../../../services/structure-attribute";
import { TypescriptStrategy } from "../../../../ts/strategy";

export const setVtlAttribute = (attribute: StructureAttribute) => {
  const strategy = TypescriptStrategy.of(attribute.project);
  const generator = attribute.generator;
  const name = strategy.formatAttributeName(attribute.name);
  const shortName = strategy.formatAttributeName(attribute.shortName);

  // non-generated
  if (generator === ResourceAttributeGenerator.NONE) {
    return `"${shortName}": $util.dynamodb.toDynamoDBJson($ctx.args.input.${name}),`;
    // generated values
  } else {
    switch (generator) {
      case ResourceAttributeGenerator.CURRENT_DATE_TIME_STAMP:
        return `"${shortName}": "$util.time.nowISO8601()"`;
      case ResourceAttributeGenerator.GUID:
        return `"${shortName}": "$util.autoId()"`;
      case ResourceAttributeGenerator.TYPE:
        return `"${shortName}": "unknown"`;

      default:
        throw new Error(`Unknown generator: ${generator}`);
    }
  }
};
