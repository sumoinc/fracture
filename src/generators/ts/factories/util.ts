import { ValueOf } from "type-fest";
import { ResourceAttributeType } from "../../../services";

export const JsType = {
  STRING: "string",
  NUMBER: "number",
  BOOLEAN: "boolean",
  ARRAY: "array",
  MAP: "map",
  OTHER: "other",
} as const;

export const jsType = (
  resourceAttributeType: ValueOf<typeof ResourceAttributeType> | string
) => {
  switch (resourceAttributeType) {
    case ResourceAttributeType.GUID:
    case ResourceAttributeType.STRING:
    case ResourceAttributeType.EMAIL:
    case ResourceAttributeType.PHONE:
    case ResourceAttributeType.URL:
    case ResourceAttributeType.DATE:
    case ResourceAttributeType.TIME:
    case ResourceAttributeType.DATE_TIME:
    case ResourceAttributeType.JSON:
    case ResourceAttributeType.IPADDRESS:
      return JsType.STRING;
    case ResourceAttributeType.INT:
    case ResourceAttributeType.FLOAT:
    case ResourceAttributeType.TIMESTAMP:
    case ResourceAttributeType.COUNT:
    case ResourceAttributeType.AVERAGE:
    case ResourceAttributeType.SUM:
      return JsType.NUMBER;
    case ResourceAttributeType.BOOLEAN:
      return JsType.BOOLEAN;
    case ResourceAttributeType.ARRAY:
      return JsType.ARRAY;
    case ResourceAttributeType.MAP:
      return JsType.MAP;
    default:
      return JsType.OTHER;
  }
};
