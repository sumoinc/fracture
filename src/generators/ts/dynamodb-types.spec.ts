import { DynamoDbTypes } from "./dynamodb-types";
import { complexResource } from "./types.spec";
import { Operation } from "../../services";
import { synthFile } from "../../util/test-util";

test("Resource formatting test", () => {
  const resource = complexResource();
  new DynamoDbTypes(resource.service);
  const content = synthFile(resource.service, "src/dynamodb-types.ts");
  expect(content).toBeTruthy();
  expect(content).toMatchSnapshot();
  //console.log(content);
});

describe("Operations", () => {
  test("Create", () => {
    const resource = complexResource();
    /*const operation = */ Operation.create(resource);
    new DynamoDbTypes(resource.service);

    const content = synthFile(resource.service, "src/dynamodb-types.ts");
    expect(content).toBeTruthy();
    expect(content).toMatchSnapshot();
    // console.log(content);
    // console.log(JSON.stringify(operation.config(), null, 2));
  });
});
