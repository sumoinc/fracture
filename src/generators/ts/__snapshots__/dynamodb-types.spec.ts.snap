// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource formatting test 1`] = `
"// ~~ Generated by projen. To modify, edit .projenrc.js and run \\"npx projen\\".

/**
 * Public facing data structure for this record.
*/
export type PersonMarshalled = {
    /**
     * Partition Key for this record.
    */
    pk: {
        S: string;
    };
    /**
     * Sort Key for this record.
    */
    sk: {
        S: string;
    };
    /**
     * Lookup value for this record.
    */
    idx: {
        S: string;
    };
    /**
     * Identifier for this record.
    */
    id: {
        S: string;
    };
    /**
     * Type of record.
    */
    t: {
        S: string;
    };
    /**
     * Version for record.
    */
    v: {
        S: string;
    };
    /**
     * The timestamp representing when this record was created.
    */
    ct: {
        S: string;
    };
    /**
     * The timestamp representing when this record was last modified.
    */
    mt: {
        S: string;
    };
    /**
     * The timestamp representing when this record was marked as deleted.
    */
    dt: {
        S: string;
    };
    fn: {
        S: string;
    };
    ln: {
        S: string;
    };
    ht: {
        N: number;
    };
    /**
     * Is this person a frog?
    */
    if: {
        BOOL: boolean;
    };
    /**
     * List opf favorite colors
    */
    favColors: {
        L: Array<Partial<StringMarshalled>>;
    };
    oneClub: {
        \\"M\\": ClubMarshalled;
    };
    ad: {
        L: Array<Partial<ClubMarshalled>>;
    };
};

/**
 * Public facing data structure for this record.
*/
export type ClubMarshalled = {
    /**
     * Partition Key for this record.
    */
    pk: {
        S: string;
    };
    /**
     * Sort Key for this record.
    */
    sk: {
        S: string;
    };
    /**
     * Lookup value for this record.
    */
    idx: {
        S: string;
    };
    /**
     * Identifier for this record.
    */
    id: {
        S: string;
    };
    /**
     * Type of record.
    */
    t: {
        S: string;
    };
    /**
     * Version for record.
    */
    v: {
        S: string;
    };
    /**
     * The timestamp representing when this record was created.
    */
    ct: {
        S: string;
    };
    /**
     * The timestamp representing when this record was last modified.
    */
    mt: {
        S: string;
    };
    /**
     * The timestamp representing when this record was marked as deleted.
    */
    dt: {
        S: string;
    };
    name: {
        S: string;
    };
};

export type CreatePersonInputMarshalled = {
    fn: {
        S: string;
    };
    ln: {
        S: string;
    };
    ht: {
        N: number;
    };
    /**
     * Is this person a frog?
    */
    if: {
        BOOL: boolean;
    };
    /**
     * List opf favorite colors
    */
    favColors: {
        L: Array<Partial<StringMarshalled>>;
    };
    oneClub: {
        \\"M\\": ClubMarshalled;
    };
    ad: {
        L: Array<Partial<ClubMarshalled>>;
    };
};

/**
 * Public facing data structure for this record.
*/
export type PersonUnmarshalled = {
    /**
     * Partition Key for this record.
    */
    pk: string;
    /**
     * Sort Key for this record.
    */
    sk: string;
    /**
     * Lookup value for this record.
    */
    idx: string;
    /**
     * Identifier for this record.
    */
    id: string;
    /**
     * Type of record.
    */
    t: string;
    /**
     * Version for record.
    */
    v: string;
    /**
     * The timestamp representing when this record was created.
    */
    ct: string;
    /**
     * The timestamp representing when this record was last modified.
    */
    mt: string;
    /**
     * The timestamp representing when this record was marked as deleted.
    */
    dt: string;
    fn: string;
    ln: string;
    ht: number;
    /**
     * Is this person a frog?
    */
    if: boolean;
    /**
     * List opf favorite colors
    */
    favColors: Array<Partial<StringUnmarshalled>>;
    oneClub: Club;
    ad: Array<Partial<ClubUnmarshalled>>;
};

/**
 * Public facing data structure for this record.
*/
export type ClubUnmarshalled = {
    /**
     * Partition Key for this record.
    */
    pk: string;
    /**
     * Sort Key for this record.
    */
    sk: string;
    /**
     * Lookup value for this record.
    */
    idx: string;
    /**
     * Identifier for this record.
    */
    id: string;
    /**
     * Type of record.
    */
    t: string;
    /**
     * Version for record.
    */
    v: string;
    /**
     * The timestamp representing when this record was created.
    */
    ct: string;
    /**
     * The timestamp representing when this record was last modified.
    */
    mt: string;
    /**
     * The timestamp representing when this record was marked as deleted.
    */
    dt: string;
    name: string;
};

export type CreatePersonInputUnmarshalled = {
    fn: string;
    ln: string;
    ht: number;
    /**
     * Is this person a frog?
    */
    if: boolean;
    /**
     * List opf favorite colors
    */
    favColors: Array<Partial<StringUnmarshalled>>;
    oneClub: Club;
    ad: Array<Partial<ClubUnmarshalled>>;
};
"
`;
